docs/architecture/self-hosted/upgrade-backup-restore.md

```diff
@@ -0,0 +1,168 @@
+---
+title: Upgrades and backups
+id: upgrade-backup-restore
+description: Learn how to upgrade, back up, and restore Prophecy.
+sidebar_position: 4
+tags:
+  - upgrade
+  - backup
+  - restore
+  - self-managed
+---
+
+This page outlines different actions you may perform to maintain your Prophecy installation.
+
+## Upgrade
+
+To upgrade Prophecy, you can do so in **Settings > Admin**.
+
+1. Navigate to the **Admin** tab of the Prophecy **Settings** page.
+2. Click on **Upgrade Version**.
+3. Choose the version you would like to upgrade to in the **Version** dropdown.
+4. Make sure that **Disable Rollback** toggle is on.
+
+![Upgrade version in Admin settings](img/upgrade-version.png)
+
+:::note
+You may want to back up Prophecy before you upgrade.
+:::
+
+## Backup
+
+Backups can be triggered manually via the API, or they can be configured to run automatically.
+
+### On-demand backups
+
+You can use the Backup API to start a backup. See [Generate API Key](./generate-api-key/) if you need an API key.
+
+Example:
+
+```
+curl --location --request POST 'https://{prophecy-url}/api/backup' \
+--header 'Cookie: prophecy-token={api-key}' \
+--header 'Content-Type: application/json' \
+--data-raw '{}'
+```
+
+Response:
+
+```
+{
+"code": 202,
+"message": "Request Accepted with timestamp 2023-02-02t16-00-00"
+}
+```
+
+### Regular automatic backups
+
+You can configure regular backups in Prophecy in settings.
+
+1. Log in to Prophecy as an admin user.
+2. Click on the three dots at the bottom left corner.
+3. Navigate to **Settings > Admin > Config > backupConfig**.
+4. Make sure `enableRegularBackups` is set to true.
+5. Edit other variables to fit your requirements.
+6. Save your changes.
+
+Below is a list of supported variables that you can change.
+
+| Configuration variable name | Description                                                                                                                                                                               | Default value |
+| --------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
+| `backupFrequency`           | How frequently to purge old user events from the internal database. Defaults to daily at 00:00. Uses [6-digit CRON](https://pkg.go.dev/github.com/robfig/cron#hdr-CRON_Expression_Format) | `0 0 0 * * *` |
+| `backupRetentionCount`      | Number of last `N` backups to retain.                                                                                                                                                     | `30`          |
+| `enableRegularBackups`      | State of automated backup creation.                                                                                                                                                       | `false`       |
+
+### Additional backup APIs
+
```

Comment:
I think we should provide at least one sample call format like 

```
 curl --location --request POST 'https://{prophecy-url}/api/backup' \
 --header 'Cookie: prophecy-token={api-key}' \
 --header 'Content-Type: application/json' \
 --data-raw '{}'
 ```

So as to know the format in which the call is to be invoked. The token here is an important one.